integration/helpers/workdir.ts

```diff
@@ -0,0 +1,50 @@
+import * as os from "node:os";
+import * as path from "node:path";
+import * as fs from "node:fs/promises";
+import * as time from "./time";
+
+export class Workdir {
+  root: string;
+  leaf: string;
+
+  constructor(leaf: string, root: string = os.tmpdir()) {
+    this.leaf = leaf;
+    this.root = path.resolve(root);
+  }
+
+  path() {
+    return path.join(this.root, this.leaf);
+  }
+
+  async create(): Promise<string> {
+    await fs.mkdir(this.path(), { recursive: true });
+    return this.path();
+  }
+
+  async exists() {
```

Comment:
See comment about using `fs.existsSync()` instead. Even if we can't use this, I think it's a duplicate implementation from `file.ts`.